<!DOCTYPE HTML>
<html lang="en">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="From sequence to structure, Chaoqun Yin">
    <meta name="description" content="Veni, Vidi, Vici">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>From sequence to structure | Chaoqun Yin</title>
    <link rel="icon" type="image/png" href="/favicon.png">

    <link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="/css/matery.css">
    <link rel="stylesheet" type="text/css" href="/css/my.css">

    <script src="/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.3.0"></head>




<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">Chaoqun Yin</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>Index</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>Tags</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>Categories</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>Archives</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>About</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>Contact</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/friends" class="waves-effect waves-light">
      
      <i class="fas fa-address-book" style="zoom: 0.6;"></i>
      
      <span>Friends</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="Search" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">Chaoqun Yin</div>
        <div class="logo-desc">
            
            Veni, Vidi, Vici
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			Index
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			Tags
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			Categories
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			Archives
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			About
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			Contact
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/friends" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-address-book"></i>
			
			Friends
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/blinkfox/hexo-theme-matery" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/blinkfox/hexo-theme-matery" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/1.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">From sequence to structure</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        padding: 35px 0 15px 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        padding-bottom: 30px;
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Computational-Biology-and-Bioinformatics/">
                                <span class="chip bg-color">Computational Biology and Bioinformatics</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/Notes/" class="post-category">
                                Notes
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>Publish Date:&nbsp;&nbsp;
                    2021-03-04
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>Update Date:&nbsp;&nbsp;
                    2021-03-19
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>Word Count:&nbsp;&nbsp;
                    2.5k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>Read Times:&nbsp;&nbsp;
                    15 Min
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>Read Count:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">

        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <h1 id="from-sequence-to-structure---secondary-structure-prediction">From sequence to structure - Secondary structure prediction</h1>
<p><a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/wim_vranken_vub_be/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen%2FAlgorithms%20in%20bioinfo%5Fcomputational%20biology%2D20210304%5F100320%2DMeeting%20Recording%2Emp4&amp;parent=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen&amp;originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOnY6L2cvcGVyc29uYWwvd2ltX3ZyYW5rZW5fdnViX2JlL0VWakxIeVdwX2F0SG1rZV9VMVNRZGdNQjBYNDNXMl9zWHR2bmVQbHFxWXIyZ0E_cnRpbWU9aVpuUjFqN2YyRWc">Video Part 1</a></p>
<h2 id="protein-secondary-structure-prediction-600"><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein_secondary_structure">Protein secondary structure</a> prediction 6:00</h2>
<p><img src="/images/ADVMB/2002180.png" /></p>
<h2 id="why-proteins-834">Why <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein">proteins</a>? 8:34</h2>
<ul>
<li><strong>Interacting</strong> is what proteins do!
<ul>
<li>Proteins can interact with many types of molecules, including <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein–protein_interaction">with other proteins</a>, <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein–lipid_interaction">with lipids</a>, <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein–carbohydrate_interaction">with carboyhydrates</a>, and <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein–DNA_interaction">with DNA</a>.</li>
</ul></li>
<li>Abundance in cells: Proteins in the cell are <strong>vastly complicated</strong>
<ul>
<li>It has been estimated that average-sized <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Bacteria">bacteria</a> contain about 2 million proteins per cell.</li>
</ul></li>
</ul>
<h2 id="bioinformatics-dna-and-proteins-1220">Bioinformatics, DNA and proteins 12:20</h2>
<table>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">total</th>
<th style="text-align: right;">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Evidence at protein level</td>
<td style="text-align: right;">20622</td>
<td style="text-align: right;">0.05%</td>
</tr>
<tr class="even">
<td>Evidence at transcript level</td>
<td style="text-align: right;">818805</td>
<td style="text-align: right;">1.91%</td>
</tr>
<tr class="odd">
<td>Inferred from homology</td>
<td style="text-align: right;">9893252</td>
<td style="text-align: right;">23.10%</td>
</tr>
<tr class="even">
<td>Predicted</td>
<td style="text-align: right;">32089200</td>
<td style="text-align: right;">74.94%</td>
</tr>
</tbody>
</table>
<p>The key method to investigate proteins is understand its level, like how to catalyse reaction and which signa, that is related to structural biology.</p>
<h2 id="structural-biology-1538"><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Structural_biology#:~:text=Structural%20biology%20is%20a%20branch%20of%20molecular%20biology,,how%20alterations%20in%20their%20structures%20affect%20their%20function.">Structural biology</a> 15:38</h2>
<figure>
<img src="/images/ADVMB/2002182.png" alt="" /><figcaption>21:24</figcaption>
</figure>
<h3 id="what-do-we-want-to-find-out-2303">What do we want to find out? 23:03</h3>
<ul>
<li><p>How does a protein <strong>interact with a ligand</strong>?</p>
<ul>
<li><p>What does <a target="_blank" rel="noopener" href="https://nl.wikipedia.org/wiki/Ibuprofen">ibuprofen</a> do?</p>
<p><img src="/images/ADVMB/2002181.png" /></p></li>
</ul></li>
<li><p>What do protein complexes look like?</p>
<ul>
<li>E.g. How do pili from bacteria form?</li>
</ul></li>
</ul>
<h3 id="protein-behaviour-2629"><a target="_blank" rel="noopener" href="https://alevelbiology.co.uk/notes/proteins-physical-chemical-properties/">Protein behaviour</a> 26:29</h3>
<p>Everything constructed proteins move all the time.</p>
<figure>
<img src="/images/ADVMB/2002183.png" alt="" /><figcaption>28:00</figcaption>
</figure>
<figure>
<img src="/images/ADVMB/2002184.png" alt="" /><figcaption>29:27</figcaption>
</figure>
<h4 id="structural-relationship-of-proteins-and-amino-acids"><a target="_blank" rel="noopener" href="https://www.dnatube.com/video/833/Structural-relationship-of-proteins-and-amino-acids">Structural relationship of proteins and amino acids</a></h4>
<p>Amino acids are the building blocks of polypeptides and proteins. In polypeptides and proteins, amino acids molecules are linked to each other by polypeptide bonds. The sequence of amino acids in polypeptide is encoded by the DNA.</p>
<h2 id="sequence-and-structure-4140"><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein_sequencing">Sequence</a> and <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein_structure">structure</a> 41:40</h2>
<figure>
<img src="/images/ADVMB/2002185.png" alt="" /><figcaption>38:17 ~ 39:41</figcaption>
</figure>
<ul>
<li><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Protein_folding">Protein folding</a></p></li>
<li><p>The structure is encoded in the sequence, but how?</p>
<ul>
<li>100 amino acid protein
<ul>
<li>198 backbone angles</li>
<li>3 conformations each</li>
<li>3<sup>198</sup> possibilities!</li>
</ul></li>
</ul></li>
<li><p>How do we get from a sequence to secondary or tertiary structure? 47:17</p>
<ul>
<li>Experimental data!
<ul>
<li>NMR, X-ray, …</li>
<li>But what if you don’t have any? – <strong>Bioinformatics</strong>!</li>
</ul></li>
</ul></li>
</ul>
<figure>
<img src="/images/ADVMB/2002186.png" alt="" /><figcaption>Bioinformatics approach 1:06:29</figcaption>
</figure>
<figure>
<img src="/images/ADVMB/2002187.png" alt="" /><figcaption>1:09:41</figcaption>
</figure>
<h2 id="protein-secondary-structure-prediction-11032">Protein secondary structure prediction 1:10:32</h2>
<h3 id="how-do-we-define-secondary-structure-classes">How do we define <em>secondary structure classes</em>?</h3>
<ul>
<li><p><a target="_blank" rel="noopener" href="https://www.youtube.com/watch?v=aH2IbYs_XjY">Hydrogen bonding</a> 1:10:48</p>
<ul>
<li>Hydrogen on electronegative atom (N, O) has partial positive charge</li>
<li>Can form a <em>hydrogen bond</em> with another electronegative atom</li>
</ul></li>
<li><p>Regular patterns of hydrogen bonding in secondary structure elements 1:16:35</p>
<ul>
<li><img src="/images/ADVMB/2002188.png" /></li>
</ul></li>
<li><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/DSSP_(hydrogen_bond_estimation_algorithm)">DSSP (hydrogen bond estimation algorithm)</a> 1:17:54</p>
<ul>
<li><p>1983, in Pascal</p></li>
<li><p>Simple definition for backbone hydrogen bond</p>
<p><img src="/images/ADVMB/2002114.png" /></p>
<ul>
<li>r is distance</li>
<li>N, H backbone atoms from residue i</li>
<li>O, C backbone atoms from residue j</li>
</ul></li>
<li><p>Assigns secondary structure based on patterns</p>
<ul>
<li>α helix (H), i → i+4</li>
<li>β strand (E), long set of connected hydrogen bonds</li>
<li>3<sub>10</sub> helix (G), i → i+3</li>
<li>And others…</li>
</ul></li>
</ul></li>
<li><p>Look at the local conformation of the amino acid.</p></li>
<li><figure>
<img src="/images/ADVMB/2002189.png" alt="" /><figcaption>1:20:48</figcaption>
</figure></li>
<li><p>Using structural biology data</p>
<ul>
<li><img src="/images/ADVMB/2002190.png" title="fig:" alt="1:27:30" /></li>
</ul></li>
</ul>
<h3 id="is-there-bias-in-the-input-data-14307">Is there bias in the input data? 1:43:07</h3>
<ul>
<li><p>Reference structure data set is crucial</p>
<ul>
<li>Not too much overlap between sequences!</li>
</ul></li>
<li><p>In general, proteins with less than 25-30% sequence identity between them</p></li>
<li><p>Other considerations</p>
<ul>
<li>Structure quality and origin, …</li>
</ul></li>
<li><p>Only about 6000 out of 80000 proteins</p></li>
<li><p>Define secondary structure</p>
<ul>
<li><p>DSSPH</p>
<ul>
<li>bonds</li>
</ul></li>
<li><p>STRIDE</p>
<ul>
<li>H bonds and (Φ,Ψ)</li>
</ul></li>
<li><p>DEFINE</p>
<ul>
<li>Interatomic distances</li>
</ul></li>
</ul></li>
<li><p>Secondary structure is a human concept!</p></li>
<li><p>Atom coordinate information from tertiary structure is converted into <strong>discrete</strong> secondary structure states</p></li>
<li><p>This is what is used to develop secondary structure predictors.</p></li>
</ul>
<p><img src="/images/ADVMB/2002191.png" /></p>
<p><a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/wim_vranken_vub_be/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen%2FAlgorithms%20in%20bioinfo%5Fcomputational%20biology%2D20210311%5F100144%2DMeeting%20Recording%2Emp4&amp;parent=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen&amp;originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOnY6L2cvcGVyc29uYWwvd2ltX3ZyYW5rZW5fdnViX2JlL0VlQ2JydHBTTHRGQ2tQclFhV0Jhd3hJQnc2UnVQWXdoc0dVdzBNQ0tPY0pTX2c_cnRpbWU9TEZoQTM2UGsyRWc">Video Part 2 1:02 ~ 1:50:17</a></p>
<h3 id="how-do-we-prepare-and-clean-up-the-data">How do we prepare and clean up the data?</h3>
<h4 id="input-data-227">Input data 2:27</h4>
<ul>
<li>Need clean protein structure data with bias removed as much as possible
<ul>
<li>What about sequence overlap? Relatively common problem, does someone already have a solution</li>
</ul></li>
<li>Look for servers/articles/… – Common bioinformatics approach 3:56
<ul>
<li>With a bit of luck the data are still live and accessible</li>
</ul></li>
<li>Determine which cutoffs/dataset to use for your approach 5:25
<ul>
<li>Very strict:
<ul>
<li>Only X-ray crystallography</li>
<li>Resolution of 1.6 Å or better (lower value)</li>
<li>Sequence identity cutoff of 25%</li>
<li>cullpdb_pc20_res1.6_R0.25_d120323_chains1998</li>
</ul></li>
</ul></li>
<li>What is in the PISCES file? 6:27</li>
</ul>
<p><img src="/images/ADVMB/2002192.png" /></p>
<ul>
<li>Run DSSP and STRIDE on 498 PDB structure 8:16
<ul>
<li>Original set reduced for computational reasons</li>
</ul></li>
<li>Parse and interpret results
<ul>
<li>Files for DSSP and for STRIDE</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002193.png" /></p>
<ul>
<li>Associated information 9:26</li>
<li>Protein family, from CATH</li>
</ul>
<p><img src="/images/ADVMB/2002194.png" /></p>
<h3 id="other-issues-1139">Other issues 11:39</h3>
<h4 id="proteins-are-dynamic">Proteins are dynamic</h4>
<h5 id="example-1222"><strong>Example</strong> 12:22:</h5>
<p>Does a given amino acid always fully adopt one type of secondary structure in reality?</p>
<ul>
<li>No!</li>
<li>Even possible that it switches secondary structure
<ul>
<li>E.g. plaque formation in Alzheimers, where α helices from native protein form βstrands in plaques.</li>
</ul></li>
</ul>
<h5 id="could-you-avoid-using-structures-1547">Could you avoid using structures? 15:47</h5>
<ul>
<li><p>Develop predictions directly from data for protein in solution</p></li>
<li><p><strong>NMR</strong> data is recorded on proteins in solution</p>
<ul>
<li>Estimate in-solution secondary structure populations from NMR observations</li>
<li>Get secondary structure fractions, not states…</li>
</ul></li>
</ul>
<h3 id="variability-1650">Variability! 16:50</h3>
<figure>
<img src="/images/ADVMB/2002195.png" alt="" /><figcaption>Many combinations and variations!</figcaption>
</figure>
<p>How will we validate how well our approach works? 18:02</p>
<h4 id="validation-2035">Validation 20:35</h4>
<ul>
<li>Understand how well the method works
<ul>
<li>Plan ahead</li>
</ul></li>
<li>Need independent test set
<ul>
<li>Cannot use data that is included in statistical analysis or training of method</li>
</ul></li>
<li>Jackknife (leave-one-out) 23:50
<ul>
<li>If method can be primed fast enough, leave out one protein, train the method, test it on the protein that you took out, ...</li>
</ul></li>
<li>But what measures or scores do you use to assess how well your method works? 25:03</li>
<li>Do you look at only what you predict correctly? Or also at what you predict incorrectly?</li>
</ul>
<h2 id="protein-secondary-structure-prediction">Protein secondary structure prediction</h2>
<h3 id="q3-measure-3535">Q<sub>3</sub> measure 35:35</h3>
<p><span class="math display">\[
Q_3 = \frac{N_{residues\_correctly\_predicted}}{N_{residues\_total}}
\]</span></p>
<ul>
<li>Random prediction should give 0.33</li>
<li>Real random prediction gives 0.38
<ul>
<li>Frequencies of secondary structure are not the same</li>
</ul></li>
<li>Bad predictions can give <strong>high</strong> scores ...</li>
</ul>
<p><img src="/images/ADVMB/2002196.png" /></p>
<h3 id="matthews-correlation-coefficient-mcc-4048">Matthews correlation coefficient (MCC) 40:48</h3>
<ul>
<li>Predicted for each secondary structure state</li>
<li>Uses false/true positives and false/true negative</li>
</ul>
<p><img src="/images/ADVMB/2002197.png" /> <span class="math display">\[
MCC = \frac{TP×TN-TP×FN}{\sqrt{(TP+FP)(TP+FN)(TN+FP)(TN+FN)}}
\]</span></p>
<ul>
<li><p>Sensitivity (hit rate, recall)</p>
<p><em>TPR = TP / (TP + FN)</em></p></li>
<li><p>Specificity</p>
<p><em>SPC = TN / (FP + TN)</em></p></li>
</ul>
<figure>
<img src="/images/ADVMB/2002198.png" alt="" /><figcaption>Always has to add up to number of residues!</figcaption>
</figure>
<ul>
<li>Report MCC always <strong><u>per secondary structure class</u></strong></li>
<li>Multi-class definition also available</li>
</ul>
<p><span class="math display">\[
MCC_{multiclass} = \frac{\sum_k\sum_l\sum_mC_{kk}C_{lm}-C_{kl}C_{mk}}{\sqrt{\sum_k(\sum_lC_{kl})(\sum_{k&#39;|k&#39;≠k}\sum_{l&#39;}C_{k&#39;l&#39;})}\sqrt{\sum_k(\sum_lC_{lk})(\sum_{k&#39;|k&#39;≠k}\sum_{l&#39;}C_{l&#39;k&#39;})}}
\]</span></p>
<h3 id="sov-measure-5029">Sov measure 50:29</h3>
<p>Segment-based, heuristic</p>
<ul>
<li>Punishes broken predictions</li>
</ul>
<p><img src="/images/ADVMB/2002199.png" /></p>
<h3 id="predictions-5330">Predictions 53:30</h3>
<figure>
<img src="/images/ADVMB/2002200.png" alt="" /><figcaption>A bit of history… how predictions evolve</figcaption>
</figure>
<ul>
<li>Model based
<ul>
<li>Preferably using some actual experimental data</li>
</ul></li>
<li>Machine learning
<ul>
<li>Many approaches, training steps essentia</li>
</ul></li>
</ul>
<h4 id="chou-fasman-5442">Chou-Fasman 54:42</h4>
<ul>
<li>Original attempt at predicting secondary structure</li>
<li>Chou and Fasman (1970s)</li>
<li>Based on very limited number of protein structures</li>
</ul>
<p>Statistical propensity for each amino acid (R) to adopt a certain secondary structure (S)</p>
<p><img src="/images/ADVMB/2002201.png" /></p>
<h5 id="chou-fasman-algorithm">Chou-Fasman algorithm</h5>
<ul>
<li>Assign α/β propensities to each amino acid</li>
</ul>
<p><img src="/images/ADVMB/2002202.png" /></p>
<ul>
<li>Find region with 4/6 connected residues p<sub>α</sub> &gt; 1.0
<ul>
<li>Extend on both ends until:
<ul>
<li>4 connected residues have average p<sub>α</sub> &lt; 1.0 → End of α helix</li>
</ul></li>
<li>Total segment longer than 5 residues, average p<sub>α</sub> &gt; p<sub>β</sub> ?
<ul>
<li>Assign as α helix</li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002203.png" /></p>
<ul>
<li>Find region with 3/5 connected residues p<sub>β</sub> &gt; 1.0
<ul>
<li>Extend on both ends until:
<ul>
<li>4 connected residues have average p<sub>β</sub> &lt; 1.0 → End of β sheet</li>
</ul></li>
<li>Average p<sub>β</sub> &gt; 1.05 and average p<sub>β</sub> &gt; p<sub>α</sub>?
<ul>
<li>Assign as β sheet</li>
</ul></li>
</ul></li>
<li>Repeat for whole sequence</li>
<li>If α/β overlap, choose which one by highest average p</li>
</ul>
<p>Overall Chou-Fasman algorithm:</p>
<ul>
<li>Based on statistics</li>
<li>Very simple
<ul>
<li>Per-residue, no residue interactions included</li>
<li>Algorithm <strong>cutoffs</strong> are pretty arbitrary</li>
</ul></li>
<li>But gives better than random results!
<ul>
<li>Q<sub>3</sub> score of 42-52% depending on protein family</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002204.png" /></p>
<h4 id="gor-method-11122">GOR method 1:11:22</h4>
<ul>
<li><p>Evolved from GOR to GOR-V</p>
<ul>
<li><em>Jean Garnier, David Osguthorpe, Barry Robson</em></li>
</ul></li>
<li><p>Based on information theory approach (Bayes)</p>
<ul>
<li>Self-information</li>
<li>Directional information</li>
<li>Pair information</li>
</ul></li>
<li><p>Clearly defined model to get to results</p></li>
<li><p>Self-information (related to Chou-Fasman)</p></li>
</ul>
<p><img src="/images/ADVMB/2002205.png" /></p>
<ul>
<li>Self-information, more general</li>
</ul>
<p><img src="/images/ADVMB/2002206.png" /></p>
<p><em>Calculated for the three conformations, the highest value of Eq. (2) for one of the conformations S will be the predicted conformation and will be the propensity for that residue to be in that conformation, usually expressed in centinat units when natural logarithms are used.</em></p>
<ul>
<li>Include local sequence information
<ul>
<li>How often do you see a residue in relation to a secondary structure element at position j?</li>
</ul></li>
</ul>
<p><span class="math display">\[
I(ΔS_j;R_1,..., R_n ) = \log[P(S_j
, R_1,..., R_n )/ P(n − S_j
, R_1,..., R_n )]
+log[P(n − S)/ P(S)]
\]</span></p>
<h5 id="gor-iii-12122">GOR I/II 1:21:22</h5>
<h6 id="gor-i-and-ii-approximation">GOR I and II approximation</h6>
<ul>
<li>Self and directional information
<ul>
<li>26 proteins, ~4500 residues (1978, GOR I)</li>
</ul></li>
<li>Uses windows of 17 residues
<ul>
<li>For residue at position j, consider j-8 to j+8</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002207.png" /></p>
<ul>
<li>GOR I and II implementation</li>
</ul>
<p><img src="/images/ADVMB/2002208.png" /></p>
<p><img src="/images/ADVMB/2002209.png" /></p>
<p><img src="/images/ADVMB/2002210.png" /></p>
<ul>
<li>GOR I results…
<ul>
<li>Has Q<sub>3</sub> around 49%, but includes turns</li>
<li>Bad at β sheet, tweaks get higher Q<sub>3</sub> (up to 60%)</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002211.png" /></p>
<h5 id="gor-iii-13505">GOR III 1:35:05</h5>
<h6 id="gor-iii-approximation">GOR III approximation</h6>
<ul>
<li>68 proteins, 12000 residues (1987)</li>
</ul>
<p><img src="/images/ADVMB/2002212.png" /></p>
<ul>
<li>The results
<ul>
<li>GOR III has Q<sub>3</sub> around 63%</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002213.png" /></p>
<p>How to improve this method?</p>
<h5 id="gor-iv-13706">GOR IV 1:37:06</h5>
<h6 id="gor-iv-approximation">GOR IV approximation</h6>
<ul>
<li>267 proteins, 63000 residues (1996)</li>
<li>Consider information for all residue pairs</li>
</ul>
<p><img src="/images/ADVMB/2002214.png" /></p>
<ul>
<li>GOR IV implementation</li>
</ul>
<p><img src="/images/ADVMB/2002215.png" /></p>
<ul>
<li>Residue at position j in conformation S (and n-S):
<ul>
<li>For threonine T at position j-8:
<ul>
<li>Probability of histidine H at position j-7</li>
<li>Same for j-6 to j+8</li>
</ul></li>
<li>For histidine H at position j-7:
<ul>
<li>…</li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002216.png" /></p>
<ul>
<li><p>Residue at position j in conformation S (and n-S):</p></li>
<li><p>Probability threonine T at position j-8</p></li>
<li><p>Probability histidine H at position j-7</p></li>
<li><p>…</p></li>
<li><p>The results</p>
<ul>
<li>have mean Q<sub>3</sub> of 64.4%</li>
<li>Example sequence of each protein class
<ul>
<li><em>Zvelebil and Baum, Understanding Bioinformatics, Garland Science.</em></li>
<li>All α, all β, α/β and α+β</li>
</ul></li>
<li>Ranges from 39.2% (α+β) to 67.2% (α/β)</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002217.png" /></p>
<h5 id="gor-v-14506">GOR V 1:45:06</h5>
<ul>
<li>Uses multiple sequence alignment
<ul>
<li>Structure is more conserved than sequence
<ul>
<li>Insertions/deletions not as common in αhelices andβsheets, more so in loops.</li>
</ul></li>
<li>PSI-BLAST on query sequence
<ul>
<li>Get set of sequences with high identity
<ul>
<li>Cutoff 30% if lots of hits, lower to 20% if not that many</li>
</ul></li>
<li>Predict secondary structure for each sequence</li>
<li>Average predictions for each residue</li>
</ul></li>
</ul></li>
<li>Various other improvements
<ul>
<li>513 proteins, 84107 residues (2002)</li>
</ul></li>
<li>The results…
<ul>
<li>Has mean Q<sub>3</sub> of 73.5%</li>
<li>Q<sub>3</sub> ranges from 60.3% (allβ) to 84.3% (allα)</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002218.png" /></p>
<ul>
<li>Why is α ‘easier’ than β?</li>
</ul>
<p><a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/sarah_galal_mohamed_ali_abdraboh_vub_be/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fsarah%5Fgalal%5Fmohamed%5Fali%5Fabdraboh%5Fvub%5Fbe%2FDocuments%2FOpnamen%2FAlgorithms%20in%20bioinfo%5Fcomputational%20biology%2D20210318%5F100322%2DMeeting%20Recording%2Emp4&amp;parent=%2Fpersonal%2Fsarah%5Fgalal%5Fmohamed%5Fali%5Fabdraboh%5Fvub%5Fbe%2FDocuments%2FOpnamen&amp;originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOnY6L2cvcGVyc29uYWwvc2FyYWhfZ2FsYWxfbW9oYW1lZF9hbGlfYWJkcmFib2hfdnViX2JlL0VRUlgyOGxSVHRGSXN0YklCUzFvRGY4QkZmemswcmFzenBEYUtyQ0YwYmZhU2c_cnRpbWU9QnVDNi05X3EyRWc">Video Part 3</a></p>
<h2 id="secondary-structure-for-unknown-sequence">Secondary structure for unknown sequence</h2>
<h3 id="zpred">Zpred</h3>
<ul>
<li>Based on GOR
<ul>
<li>Uses homologous sequence information
<ul>
<li>Averaging of prediction per residue</li>
<li>Calculates <em>conservation value</em></li>
</ul></li>
</ul></li>
<li>Some amino acids are more similar to each other than others…
<ul>
<li>Related to physical characteristics (of the side chain), not DNA sequence mutation variation</li>
</ul></li>
<li>Amino acid characteristics
<ul>
<li><img src="/images/CS61A/2002219.png" /></li>
<li><img src="/images/CS61A/2002220.png" /></li>
</ul></li>
<li>Conservation value
<ul>
<li>Zvelebil conservation number (C<sub>n</sub>)
<ul>
<li>1.0 for identical residue</li>
<li>0.9 for residues with same characteristics (e.g. I and L)</li>
<li>Subtract 0.1 for every different characteristic
<ul>
<li>R positive, polar</li>
<li>V aliphatic, hydrophobic, small</li>
<li>5 differences; conservation 0.9 – (0.1 x 5) = 0.4</li>
</ul></li>
<li>Gaps get all characteristics – Results in low conservation value!</li>
</ul></li>
</ul></li>
<li>The results…</li>
<li>Q<sub>3</sub> ranges from 44.9% (allβ) to 76.9% (allα)</li>
<li>Results for all β are worse than GOR IV…</li>
</ul>
<h3 id="nearest-neighbour-method">Nearest-neighbour method</h3>
<ul>
<li>Proteins similar sequence share same fold
<ul>
<li>Assumption: also true for short stretches</li>
<li><img src="/images/CS61A/2002221.png" /></li>
</ul></li>
<li>Examine query sequence with same window
<ul>
<li><img src="/images/CS61A/2002222.png" /></li>
<li>Identify best-matching fragments</li>
<li>Look at most prevalent <strong>secondary structure</strong> for middle residue</li>
</ul></li>
</ul>
<h3 id="predator">PREDATOR</h3>
<ul>
<li>Identifies potentially hydrogen-bonded residues
<ul>
<li>α helices, parallel and antiparallel βsheet</li>
</ul></li>
<li>Nearest-neighbor for secondary structure
<ul>
<li>α helices, βsheet and coil</li>
</ul></li>
<li>Propensities
<ul>
<li>Turns</li>
</ul></li>
<li>Combines results</li>
<li>Can use homologous sequences
<ul>
<li>Pairwise alignments!</li>
</ul></li>
<li>The results without homologous…
<ul>
<li>Q<sub>3</sub> ranges from 49.0% (α+β) to 66.4% (α/β)</li>
<li>Similar to GOR IV</li>
</ul></li>
<li>The results with homologous…
<ul>
<li>Q<sub>3</sub> ranges from 54.2% (α+β) to 80.5% (α/β)</li>
<li>Again homologous sequences help…</li>
</ul></li>
<li>Results
<ul>
<li><img src="/images/CS61A/2002223.png" /></li>
</ul></li>
</ul>
<h3 id="hidden-markov-model">Hidden Markov model</h3>
<ul>
<li>Discrete Markov process
<ul>
<li><img src="/images/ADVMB/2002219.png" /></li>
</ul></li>
<li>Transition probabilities expressed in transition matrix
<ul>
<li><img src="/images/ADVMB/2002220.png" /></li>
</ul></li>
</ul>
<h4 id="discrete-markov-process">Discrete Markov process</h4>
<ul>
<li><p>Probability for Sun-cloud-rain-rain-sun?</p></li>
<li><p>First have to define starting state (sun), then: <span class="math display">\[
P = P(sun |sun)× P(cloud |sun)×P(rain | cloud)× P(rain |rain)× P(sun |rain)
P = 0.50 × 0.30 ×0.35× 0.40 × 0.10 = 0.0021
\]</span></p></li>
<li><p>Can get probability for sequence of related events</p>
<ul>
<li>Dependence only on previous state(s)!</li>
<li>Markov chain</li>
</ul></li>
</ul>
<h4 id="hidden-markov-model-hmm">Hidden Markov Model (HMM)</h4>
<p><img src="/images/ADVMB/2002221.png" /></p>
<ul>
<li><strong>Define</strong>:
<ul>
<li><p>Hidden states (N=3) <span class="math display">\[
S = \{S_1, S_2,..., S_N\}
\]</span></p></li>
<li><p>State of model at step n <span class="math display">\[
q_n
\]</span></p></li>
<li><p>Observations (M=20) <span class="math display">\[
V = \{v_1, v_2,..., v_m \}
\]</span></p></li>
<li><p>State transition matrix <span class="math display">\[
A = \{p_{i|j}\}\\
p_{i|j} = P(q_{n+1} = i | q_{n} = j),1≤ i, j ≤ N
\]</span></p></li>
<li><p>Emission distribution <span class="math display">\[
E = \{e_j(k)\}\\e_j(k) = P(V_kat n | q_n = S_{j});1≤ j ≤ N,1≤ k ≤ M
\]</span></p></li>
<li><p>Initial state distribution <span class="math display">\[
π= π_{i}\\
π_{i} = P(q_1 = S_j),1≤ j ≤ N
\]</span></p></li>
</ul></li>
<li>3-state secondary structure HMM
<ul>
<li>Produce observation sequence
<ul>
<li>Choose an initial state S<sub>j</sub> according to π; step q<sub>1 </sub></li>
<li>Choose an emission observation v<sub>k</sub> based on state</li>
<li>Proceed to step q<sub>n+1</sub>, pick state S<sub>i</sub> using transition probabilities from matrix A</li>
<li>Again emission, proceed or stop, …</li>
</ul></li>
</ul></li>
<li>Have to ‘recover’ the hidden state probabilities from a known observed sequence
<ul>
<li>Calculate all state sequences of sequence length T?
<ul>
<li>2TN<sup>T</sup> calculations; not efficient</li>
</ul></li>
<li>Forward algorithm
<ul>
<li>Start with first amino acid in sequence
<ul>
<li>Calculate probabilities for observing next amino acid in sequence for each secondary structure combination</li>
</ul></li>
<li>N2T calculations</li>
</ul></li>
</ul></li>
<li>Hidden states model can be made more complex to take more variables into account
<ul>
<li>Hydrophobicity, charge bias, helix length, …</li>
<li>Incorporate homologous sequence</li>
</ul></li>
<li>For example HHPred
<ul>
<li><img src="/images/ADVMB/2002222.png" /></li>
</ul></li>
</ul>
<h3 id="machine-learning">Machine learning</h3>
<ul>
<li>Neural networks</li>
<li>Random forests</li>
<li>Support vector machine</li>
<li>…</li>
</ul>
<h4 id="neural-networks">Neural networks</h4>
<ul>
<li>Layers with connected <strong>nodes</strong> or <strong>units</strong></li>
</ul>
<p><img src="/images/ADVMB/2002223.png" /></p>
<p><img src="/images/ADVMB/2002224.png" /></p>
<p><img src="/images/ADVMB/2002225.png" /></p>
<p><img src="/images/ADVMB/2002226.png" /></p>
<ul>
<li>Many architectures now possible for neural networks
<ul>
<li>Is a world in itself</li>
</ul></li>
<li>Always crucial is the network training
<ul>
<li>Training set
<ul>
<li>With as little bias as possible</li>
<li>Is used to make the neural network ‘learn’</li>
</ul></li>
</ul></li>
<li>Complexity of network to fit problem
<ul>
<li>Want predictions as ‘general’ as possible!</li>
</ul></li>
</ul>
<h5 id="network-training">Network training</h5>
<ul>
<li>Test set
<ul>
<li>To validate results and avoid overfitting</li>
</ul></li>
</ul>
<p><img src="/images/ADVMB/2002227.png" /></p>
<h4 id="phd">PHD</h4>
<ul>
<li>Uses two neural networks
<ul>
<li>First sequence-to-structure
<ul>
<li>Predict secondary structure</li>
</ul></li>
<li>Second structure-to-structure
<ul>
<li>Takes output of first network</li>
<li>Correlate conformations for neighbouring residues</li>
</ul></li>
</ul></li>
<li>The results…
<ul>
<li>Q<sub>3</sub> ranges from 73.5% (allβ) to 87.9% (allα)</li>
<li>Much better…</li>
</ul></li>
</ul>
<p>Example:</p>
<p><img src="/images/ADVMB/2002228.png" /></p>
<h4 id="prof">PROF</h4>
<ul>
<li>Predict secondary structure with different approaches
<ul>
<li>GOR-type, multiple sequence alignment, …</li>
</ul></li>
<li>Use neural network to choose final prediction</li>
<li>The results…
<ul>
<li>Q<sub>3</sub> ranges from 58.15 (allβ) to 90.7% (allα)</li>
</ul></li>
</ul>
<h4 id="psipred">PSIPRED</h4>
<ul>
<li>Similar to PHD
<ul>
<li>But with first PSI-BLAST multiple alignment step</li>
</ul></li>
<li>The results…
<ul>
<li>Q<sub>3</sub> ranges from 65.4% (allβ) to 94.4% (allα)</li>
</ul></li>
</ul>
<h4 id="hurray-for-neural-networks">Hurray for neural networks</h4>
<ul>
<li>Give the best results overall</li>
<li>Black box</li>
<li>Does not contribute to our understanding of how sequence encodes secondary structure…</li>
<li>Lots of effort going towards ‘understandable AI’
<ul>
<li>But not so easy</li>
</ul></li>
</ul>
<h3 id="meta--or-consensus-predictors">Meta- or consensus predictors</h3>
<ul>
<li>Combine results from different approaches</li>
<li>Increases confidence in positives</li>
</ul>
<h4 id="combination-methods">Combination methods</h4>
<p>For example JPred</p>
<p><img src="/images/ADVMB/2002229.png" /></p>
<p>JPred</p>
<p><img src="/images/ADVMB/2002230.png" /></p>
<h4 id="bioinformatics-approach">Bioinformatics approach</h4>
<p>Combine results from servers</p>
<ul>
<li>Own scripts or workflow managers (TAVERNA)</li>
</ul>
<p><img src="/images/ADVMB/2002231.png" /></p>
<h2 id="which-data-in-the-end">Which data in the end?</h2>
<p><img src="/images/ADVMB/2002232.png" /></p>
<h2 id="sequence-to-secondary-structure">Sequence to secondary structure</h2>
<ul>
<li>Validation
<ul>
<li>Community testing efforts (CASP, CAFASP)</li>
<li>Many approaches already tried before…</li>
<li>Always worth checking out!</li>
</ul></li>
<li>What does the input data mean?
<ul>
<li>Understand the results</li>
<li>Know what can be used as input in the first place</li>
</ul></li>
<li>What does the output of each method mean?
<ul>
<li>What are you using it for?</li>
<li>Accuracy, confidence, …</li>
</ul></li>
<li>Why again?
<ul>
<li>Enormous amount of sequence-only data – Identify structured regions:
<ul>
<li>Can give insights in possible function</li>
<li>Can help in deciding where to mutate sequence</li>
</ul></li>
</ul></li>
<li>But!
<ul>
<li>Sequences can adopt multiple secondary structure in different circumstances...
<ul>
<li>Absolute truths are rare in biology!</li>
</ul></li>
<li>Alpha-helix easier than beta-sheet…</li>
</ul></li>
<li>Can probably only get up to ~90% accuracy</li>
</ul>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        Author:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">Chaoqun Yin</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        Link:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="https://yintelligence.tech/2021/03/04/2021-03-04-From-sequence-to-structure/">https://yintelligence.tech/2021/03/04/2021-03-04-From-sequence-to-structure/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        Reprint policy:
                    </i>
                </span>
                <span class="reprint-info">
                    All articles in this blog are used except for special statements
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    reprint polocy. If reproduced, please indicate source
                    <a href="/about" target="_blank">Chaoqun Yin</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>Copied successfully, please follow the reprint policy of this article</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">more</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Computational-Biology-and-Bioinformatics/">
                                    <span class="chip bg-color">Computational Biology and Bioinformatics</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;Previous</div>
            <div class="card">
                <a href="/2021/03/05/2021-03-05-Containers/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/10.jpg" class="responsive-img" alt="Containers">
                        
                        <span class="card-title">Containers</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2021-03-05
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Notes/" class="post-category">
                                    Notes
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Structure-and-Interpretation-of-Computer-Programs/">
                        <span class="chip bg-color">Structure and Interpretation of Computer Programs</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                Next&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2021/03/04/2021-03-04-Testing/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/8.jpg" class="responsive-img" alt="Testing">
                        
                        <span class="card-title">Testing</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2021-03-04
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Notes/" class="post-category">
                                    Notes
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Data-Structures/">
                        <span class="chip bg-color">Data Structures</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>


<script>
    $('#articleContent').on('copy', function (e) {
        // IE8 or earlier browser is 'undefined'
        if (typeof window.getSelection === 'undefined') return;

        var selection = window.getSelection();
        // if the selection is short let's not annoy our users.
        if (('' + selection).length < Number.parseInt('120')) {
            return;
        }

        // create a div outside of the visible area and fill it with the selected text.
        var bodyElement = document.getElementsByTagName('body')[0];
        var newdiv = document.createElement('div');
        newdiv.style.position = 'absolute';
        newdiv.style.left = '-99999px';
        bodyElement.appendChild(newdiv);
        newdiv.appendChild(selection.getRangeAt(0).cloneContents());

        // we need a <pre> tag workaround.
        // otherwise the text inside "pre" loses all the line breaks!
        if (selection.getRangeAt(0).commonAncestorContainer.nodeName === 'PRE') {
            newdiv.innerHTML = "<pre>" + newdiv.innerHTML + "</pre>";
        }

        var url = document.location.href;
        newdiv.innerHTML += '<br />'
            + 'From: Chaoqun Yin<br />'
            + 'Author: Chaoqun Yin<br />'
            + 'Link: <a href="' + url + '">' + url + '</a><br />'
            + '本文章著作权归作者所有，任何形式的转载都请注明出处。';

        selection.selectAllChildren(newdiv);
        window.setTimeout(function () {bodyElement.removeChild(newdiv);}, 200);
    });
</script>


<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;TOC</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h1, h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h1, h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>


<script src="https://cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script>
    MathJax.Hub.Config({
        tex2jax: {inlineMath: [['$', '$'], ['\(', '\)']]}
    });
</script>



    <footer class="page-footer bg-color">
    
    <div class="container row center-align" style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2021</span>
            
            <span id="year">2021</span>
            <a href="/about" target="_blank">Chaoqun Yin</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2021";
                    var startMonth = "1";
                    var startDate = "28";
                    var startHour = "2";
                    var startMinute = "1";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/yintelligence" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:Yintelligence@outlook.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;Search</span>
            <input type="search" id="searchInput" name="s" placeholder="Please enter a search keyword"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

	
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
